test/functional/cmdLineTests/criu/src/org/openj9/criu/UsernameTest.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2023
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.criu;
+
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class UsernameTest {
+
+    private static String getUsernameFromProperty() {
+        return System.getProperty("user.name");
+    }
+
+    private static String getExpectedUsername() throws IOException {
+        // Getting the username from the USER env var is easier, but
+        // since env vars also have special treatment in checkpoint mode
+        // we don't want to rely on them to validate username handling.
+        // Instead we call the `id` system command.
+        return new BufferedReader(new InputStreamReader(Runtime.getRuntime().exec("id --user --name").getInputStream())).readLine().trim();
+    }
+
+    public static void main(String args[]) throws IOException {
+        Path path = Paths.get("cpData");
+        String expectedUsernamePreCheckpoint = getExpectedUsername();
+        String usernamePreCheckpoint = getUsernameFromProperty(); // Checkpointing allowed, should skip getpwuid and fall back to env.
+        CRIUTestUtils.checkPointJVM(path);
```

Comment:
Though it is not harmful to do a checkpoint, it is not needed. The problem exists before the checkpoint.

Reply:
Agree. I wrote the test initially without a checkpoint and it does the job.

I also wanted the test to exercise both paths (checkpoint allowed and not allowed) so adding a checkpoint and another call to `System.getProperty("user.name")` seemed like the easiest way to get better coverage.

Reply:
A couple of comments about the format:
* Java file indention uses the tab instead of spaces (`criu_nonPortable.xml` is fine).
* A long comment is usually in its own line.

Reply:
Fixed, thanks.